Release/src/http/client/http_client_asio.cpp

```diff
@@ -345,48 +346,6 @@ class asio_connection
     bool m_closed;
 };
 
-class connection_pool_stack
-{
-public:
-    // attempts to acquire a connection from the deque; returns nullptr if no connection is
-    // available
-    std::shared_ptr<asio_connection> try_acquire() CPPREST_NOEXCEPT
-    {
-        const size_t oldConnectionsSize = m_connections.size();
-        if (m_highWater > oldConnectionsSize)
```

Comment:
The bug was caused by this m_highWater modification happening with the old connection pool size, rather than the new one; this was correct when connection_pool_stack was connectionPoolQueue :)

Fixed this, extracted, and added tests.